### PR TITLE
migrate to Github Actions

### PR DESCRIPTION
**Pull request by akrherz**
Here's my first foray into a github action workflow for openfire plugins.  Please closely nitpick this and finalize something before I replicate this for all plugins and then find out I have something wrong!

### CODE REVIEW

**Comment by akrherz** [comment]
Please see [github action output](https://github.com/akrherz/openfire-sip-plugin/actions/runs/398471633)

**Comment by akrherz** [comment]
Oh, I doubt this works for deploy as `maven-settings-for-openfire-plugins.xml` has

```
      <username>${env.CI_DEPLOY_USERNAME}</username>
      <password>${env.CI_DEPLOY_PASSWORD}</password>
```
Hmm, unsure I guess.

**Comment by Fishbowler** [comment]
Deploy should be possible though. Maven username and password are already available in the Ignite Realtime secrets - I think you put them there for me to use in Openfire. https://github.com/igniterealtime/Openfire/settings/secrets/actions

**Comment by akrherz** [comment]
> Maven username and password are already available in the Ignite Realtime secrets

Thanks, what I am wondering though is what their variable names are.  I think the curernt xml file is specific to Travis nomenclature.

**Comment by Fishbowler** [comment]
Duplicate the file within ci-tooling for GH Actions, whilst we work through migrating them all?

**Review by Fishbowler — APPROVED** [review]

**Comment by akrherz** [comment]
> Duplicate the file within ci-tooling for GH Actions, whilst we work through migrating them all?

Hehe, I would get all the plugins migrated in one day.

**Comment by akrherz** [comment]
Sigh, I likely need to merge this to see if it indeed breaks...

**Comment by Fishbowler** [comment]
I had that problem with Openfire. It immediately broke.
Certainly the way you're setting the env is exactly what I'd have done to get them into scope.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        run: |
+          git clone --depth 1 https://github.com/igniterealtime/ci-tooling.git target/ci-tooling
+          cp target/ci-tooling/maven-settings-for-openfire-plugins.xml $HOME/.m2/settings.xml
+
+      - name: Build with Maven
+        run:  mvn -B package
+
+      - name: Conditionally Deploy to Igniterealtime Archiva
+        if: ${{github.repository == 'igniterealtime/openfire-sip-plugin' && github.ref == 'refs/heads/master' && github.event_name == 'push' && matrix.java == '1.8'}}
+        run: mvn deploy --settings target/ci-tooling/maven-settings-for-openfire-plugins.xml
```
Comment (Fishbowler) [inline .github/workflows/build.yml]: Needs -B to never ask for input

**Pull request merged**